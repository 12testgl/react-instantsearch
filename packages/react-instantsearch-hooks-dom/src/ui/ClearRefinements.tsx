import React from 'react';

import { cx } from './lib/cx';

export type ClearRefinementsTranslations = {
  /**
   * The label of the button
   */
  resetLabel: string;
};

type WrapperProps = React.ComponentProps<'div'>;

export type ClearRefinementsWidgetProps = WrapperProps & {
  classNames?: Partial<ClearRefinementsClassNames>;
};

export type ClearRefinementsProps = ClearRefinementsWidgetProps &
  Pick<React.ComponentProps<'button'>, 'disabled' | 'onClick'> & {
    translations: ClearRefinementsTranslations;
  };

export type ClearRefinementsClassNames = {
  /**
   * Class names to apply to the root element
   */
  root: string;
  /**
   * Class names to apply to the button
   */
  button: string;
  /**
   * Class names to apply to the button when it's disabled
   */
  disabledButton: string;
};

export function ClearRefinements({
  classNames = {},
  disabled,
  onClick,
  translations,
  ...props
}: ClearRefinementsProps) {
  return (
    <div
      {...props}
      className={cx('ais-ClearRefinements', classNames.root, props.className)}
    >
      <button
        disabled={disabled}
        onClick={onClick}
        className={cx(
          'ais-ClearRefinements-button',
          classNames.button,
          disabled &&
            cx(
              'ais-ClearRefinements-button--disabled',
              classNames.disabledButton
            )
        )}
      >
        {translations.resetLabel}
      </button>
    </div>
  );
}
