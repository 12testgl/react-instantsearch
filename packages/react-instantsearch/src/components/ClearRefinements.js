import PropTypes from 'prop-types';
import React, { Component } from 'react';
import translatable from '../core/translatable';
import createClassNames from './createClassNames';

const cx = createClassNames('ClearRefinements');

class ClearRefinements extends Component {
  static propTypes = {
    items: PropTypes.arrayOf(PropTypes.object).isRequired,
    canRefine: PropTypes.bool.isRequired,
    refine: PropTypes.func.isRequired,
    translate: PropTypes.func.isRequired,
  };

  render() {
    const { items, canRefine, refine, translate } = this.props;

    return (
      <div className={cx('')}>
        <button
          className={cx('button', !canRefine && 'button--disabled')}
          onClick={() => refine(items)}
          disabled={!canRefine}
        >
          {translate('reset')}
        </button>
      </div>
    );
  }
}

export default translatable({
  reset: 'Clear all filters',
})(ClearRefinements);
