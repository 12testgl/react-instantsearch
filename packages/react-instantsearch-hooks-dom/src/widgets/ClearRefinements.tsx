import React from 'react';
import { useClearRefinements } from 'react-instantsearch-hooks';

import { ClearRefinements as ClearRefinementsUiComponent } from '../ui/ClearRefinements';

import type { ClearRefinementsWidgetProps } from '../ui/ClearRefinements';
import type { UseClearRefinementsProps } from 'react-instantsearch-hooks';

export type ClearRefinementsProps = ClearRefinementsWidgetProps &
  UseClearRefinementsProps;

export function ClearRefinements({
  includedAttributes,
  excludedAttributes,
  transformItems,
  ...props
}: ClearRefinementsProps) {
  const { canRefine, refine } = useClearRefinements(
    {
      includedAttributes,
      excludedAttributes,
      transformItems,
    },
    {
      $$widgetType: 'ais.clearRefinements',
    }
  );

  return (
    <ClearRefinementsUiComponent
      {...props}
      translations={{
        resetLabel: 'Clear refinements',
      }}
      onClick={refine}
      disabled={!canRefine}
    />
  );
}
