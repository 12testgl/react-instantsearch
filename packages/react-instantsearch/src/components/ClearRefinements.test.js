/* eslint-env jest, jasmine */

import React from 'react';
import renderer from 'react-test-renderer';
import Enzyme, { mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
Enzyme.configure({ adapter: new Adapter() });

import ClearRefinements from './ClearRefinements';

describe('ClearRefinements', () => {
  it('renders a clickable button', () =>
    expect(
      renderer
        .create(
          <ClearRefinements
            refine={() => null}
            items={[{ filter: 1 }]}
            cx={(...x) => x.join(' ')}
          />
        )
        .toJSON()
    ).toMatchSnapshot());

  it('has a disabled mode', () =>
    expect(
      renderer
        .create(
          <ClearRefinements
            refine={() => null}
            items={[]}
            cx={(...x) => x.join(' ')}
          />
        )
        .toJSON()
    ).toMatchSnapshot());

  it('is disabled when there is no filters', () => {
    const refine = jest.fn();
    const wrapper = mount(
      <ClearRefinements refine={refine} items={[]} cx={(...x) => x.join(' ')} />
    );

    const btn = wrapper.find('button');
    expect(refine.mock.calls).toHaveLength(0);
    btn.simulate('click');
    expect(refine.mock.calls).toHaveLength(0);
  });

  it('is not disabled when there are filters', () => {
    const refine = jest.fn();
    const wrapper = mount(
      <ClearRefinements
        refine={refine}
        items={[{ value: 'test', label: 'test: test' }]}
        cx={(...x) => x.join(' ')}
      />
    );

    const btn = wrapper.find('button');
    expect(refine.mock.calls).toHaveLength(0);
    btn.simulate('click');
    expect(refine.mock.calls).toHaveLength(1);
  });
});
