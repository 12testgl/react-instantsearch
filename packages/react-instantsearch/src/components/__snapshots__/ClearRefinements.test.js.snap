// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClearRefinements has a disabled mode 1`] = `
<button
  className="button button--disabled"
  disabled={true}
  onClick={[Function]}
>
  Clear all filters
</button>
`;

exports[`ClearRefinements renders a clickable button 1`] = `
<button
  className="button false"
  disabled={false}
  onClick={[Function]}
>
  Clear all filters
</button>
`;
